src/App.tsx

```diff
@@ -15,47 +16,133 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((good1, good2) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return good1.localeCompare(good2);
+
+      case SortType.LENGTH:
+        return good1.length - good2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    return visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, reverseList] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  const displayReset = sortType !== SortType.NONE || isReversed;
```

Comment:
Naming conventions are important in programming because they make code more readable and easier to understand. Here are some common naming conventions for boolean variables and functions:

**Boolean Variables:**
- Start with “is”, “has”, “should”, or “does” to indicate that the variable represents a true or false condition.
- Use descriptive names that accurately convey the meaning of the variable.
- Use `camelCase` for multi-word names (e.g. “isCompleted” instead of “is_completed”).

Examples:
```
isLoggedin
hasPermission
shouldShowDialog
doesExist
```

**Functions:**
- Start with a verb that describes the action the function performs.
- Use descriptive names that accurately convey the purpose of the function.
- Use `camelCase` for multi-word names.

Examples:
```
handleClick
validateForm
renderListItems
deleteAccount
```

These naming conventions help to make code more readable and self-explanatory, which is especially important when working on larger projects with other developers.